### ISSUE TITLE
source should not be used as a variable as it prevents debugging.

### ISSUE THREAD

**Issue by bubthegreat** [PyCQA/pylint#1873]
### Steps to reproduce
1. Use the variable name 'source', or any other pdb function  as a local in any function.
2. Debug that function using pdb, ipdb, or any derivative.
3. Try to access the local variable and it will have been overwritten.

In [4]: def myfunc():
   ...:     source = 'string'
   ...:     import ipdb; ipdb.set_trace()
   ...:     return source
   ...: 

In [5]: myfunc()
*** SyntaxError: invalid syntax
Warning: disable autoreload in ipython_config.py to improve performance.
> <ipython-input-4-3756fd2b9ee2>(4)myfunc()
      2     source = 'string'
      3     import ipdb; ipdb.set_trace()
----> 4     return source

ipdb> locals()['source']
'string'
ipdb> source
*** SyntaxError: unexpected EOF while parsing
ipdb> 

### Current behavior
Pylint does not warn about using the keyword for pdb.source as a variable.

### Expected behavior
Pylint should warn users that they may be hindering debugging when using this variable.

**Comment by brycepg** [comment]
You could use `!source` in pdb to reference the `source` variable instead of the command

**Comment by PCManticore** [comment]
I'm not exactly sure why pylint should catch this problem.

**Comment by jsta** [comment]
I would very much like this functionality. Considering developing a plugin if it's possible.

**Comment by Pierre-Sassoulas** [comment]
@jsta this is probably as easy as adding the value in the default 'bad' names. Or maybe we could treat it like `id` or `help` but I'm not sure there's a data structure ready to fill for this (might be generic / dynamic). In any case do not hesitate to ask questions if there's an issue with the approach.

**Comment by Pierre-Sassoulas** [comment]
Just to confirm, this don't need additional development, it can be fixed with configuration:
```toml
[tool.pylint.basic]
# Bad variable names which should always be refused, separated by a comma.
bad-names = ["foo", "bar", "baz", "source"]
enable=disallowed-name
```
See https://pylint.readthedocs.io/en/latest/user_guide/messages/convention/disallowed-name.html
